packages/mui-material/src/Button/Button.js

```diff
@@ -88,83 +88,87 @@ const ButtonRoot = styled(ButtonBase, {
     ),
     '&:hover': {
       textDecoration: 'none',
-      backgroundColor: alpha(theme.palette.text.primary, theme.palette.action.hoverOpacity),
+      backgroundColor: theme.vars
+        ? `rgba(${theme.vars.palette.text.primaryChannel} / ${theme.vars.opacity?.hover})`
+        : alpha(theme.palette.text.primary, theme.palette.action.hoverOpacity),
       // Reset on touch devices, it doesn't add specificity
       '@media (hover: none)': {
         backgroundColor: 'transparent',
       },
       ...(ownerState.variant === 'text' &&
         ownerState.color !== 'inherit' && {
-          backgroundColor: alpha(
-            theme.palette[ownerState.color].main,
-            theme.palette.action.hoverOpacity,
-          ),
+          backgroundColor: theme.vars
+            ? `rgba(${theme.vars.palette[ownerState.color].mainChannel} / ${
+                theme.vars.opacity?.hover
+              })`
+            : alpha(theme.palette[ownerState.color].main, theme.palette.action.hoverOpacity),
           // Reset on touch devices, it doesn't add specificity
           '@media (hover: none)': {
             backgroundColor: 'transparent',
           },
         }),
       ...(ownerState.variant === 'outlined' &&
         ownerState.color !== 'inherit' && {
-          border: `1px solid ${theme.palette[ownerState.color].main}`,
-          backgroundColor: alpha(
-            theme.palette[ownerState.color].main,
-            theme.palette.action.hoverOpacity,
-          ),
+          border: `1px solid ${(theme.vars || theme).palette[ownerState.color].main}`,
+          backgroundColor: theme.vars
+            ? `rgba(${theme.vars.palette[ownerState.color].mainChannel} / ${
+                theme.vars.opacity?.hover
```

Comment:
@siriwatknp I just noticed that we should use here the `theme.vars.palette.action.hoverOpacity`, as it differs in light/dark mode. (0.04 vs 0.08 in dark mode). Otherwise it won't look the same as in the original MD implementation.

Reply:
We can update the styles to use the `theme.vars.opacity` in v6 I guess after revisiting all usages in different components.